contrib/babelfishpg_tsql/src/hooks.c

```diff
@@ -1290,6 +1300,9 @@ bbf_object_access_hook(ObjectAccessType access, Oid classId, Oid objectId, int s
 	if (prev_object_access_hook)
 		(*prev_object_access_hook) (access, classId, objectId, subId, arg);
 
+	if (access == OAT_DROP && classId == RelationRelationId)
+		pltsql_drop_view_definition(objectId);
```

Comment:
is this hook called when base object which a T-SQL view depends on is dropped cascade?

Reply:
yes. it will be called.
Next question might be then why `clean_up_bbf_view_def()` is needed?
Reason: 
- When we drop logical db in babelfish, `drop_bbf_db()` function gets executed. 
- In that function we generate subcommands to drop child objects.
- At that time this hook is called but in that case we're unable to lookup babelfish_namespace catalog to get logical schema name. So need to explicitly clean the catalog in case of `drop_bbf_db()`